### PR TITLE
 feat(broker): use FEEL for conditions on sequence flows

### LINKED ISSUE #3805: Advanced expression support for sequence flow conditions
**Is your feature request related to a problem? Please describe.**
For a sequence flow on an exclusive gateway, I can define the `conditionExpression` as FEEL expressions.

**Describe the solution you'd like**
Same as in #3792.

**Additional context**
Currently, the `conditionExpression` is an expression with a limited syntax (https://docs.zeebe.io/reference/conditions.html). The issue is about changing the format of the expression to FEEL.

Existing expressions need to be migrated. For example, 
`owner == "Paul"` to `owner = "Paul"` or 
`order.count >= 5 && order.count < 15` to `order.count >= 5 and order.count < 15`.

### PR DESCRIPTION
**Pull request by saig0**
## Description

* replace the existing condition language with FEEL to uniform the expression languages
* conditions can be used on sequence flows that are attached to an exclusive gateway
* remove `msgpack-el` module

## Related issues

closes #3805

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: engine/src/test/resources/workflows/invalid_process_condition.bpmn [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-  <bpmn:exclusiveGateway id="gateway" />
-  <bpmn:sequenceFlow id="flow2" sourceRef="gateway" targetRef="end" >
-    <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">foo</bpmn:conditionExpression>
-  </bpmn:sequenceFlow>
-  <bpmn:endEvent id="end" />
+    <bpmn:startEvent id="start"/>
+    <bpmn:sequenceFlow id="flow1" sourceRef="start" targetRef="gateway"/>
+    <bpmn:exclusiveGateway id="gateway"/>
+    <bpmn:sequenceFlow id="flow2" sourceRef="gateway" targetRef="end">
+      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">=?!</bpmn:conditionExpression>
```
Comment (deepthidevaki) [inline engine/src/test/resources/workflows/invalid_process_condition.bpmn]: What if the condition is `foo`? Is it a valid condition expression?

Comment (saig0) [reply engine/src/test/resources/workflows/invalid_process_condition.bpmn]: Yes, it is. `foo` could be a boolean variable that is computed before. 
Currently, we need to write 'foo == true` :sweat_smile:

Comment (deepthidevaki) [reply engine/src/test/resources/workflows/invalid_process_condition.bpmn]: So it is always treated as an expression to be evaluated, not just a static value foo.

Comment (saig0) [reply engine/src/test/resources/workflows/invalid_process_condition.bpmn]: Correct :+1: 
I see no use case for a static value here.

**Review by deepthidevaki — APPROVED** [review]
:+1:

**Comment by saig0** [comment]
RIP `msgpack-el` module :wastebasket: (~2k LoC)

**Comment by saig0** [comment]
bors retry

**Comment by Zelldon** [comment]
> RIP msgpack-el module wastebasket (~2k LoC)

@saig0 :+1:

**Comment by saig0** [comment]
`RestoreTest.shouldKeepPositionsConsistent` failed again.

**Pull request merged**